### PR TITLE
docs: contained list enhancements

### LINKED ISSUE #3198: docs: add guidance to contained list
stub

### PR DESCRIPTION
**Pull request by laurenmrice**
Closes #3198 

- Adds additional content and image guidance on the usage and style tabs for contained list enhancements.

- **NOTE:** Do not merge until enhancements are in code.
  -  [x] [PR: add sizes, interactive element examples](https://github.com/carbon-design-system/carbon/pull/12621#event-7858096370) 
  - [x] [PR: persistent search](https://github.com/carbon-design-system/carbon/pull/13442)
  - [x] [PR: inset rule dividers](https://github.com/carbon-design-system/carbon/pull/12795)

---

#### Changelog

**New**

- Images for new modifiers and size options.
- Written content for new modifiers and size options.

**Changed**

- Update most images to show extended rules as the default option.

### CODE REVIEW

**Comment by laurenmrice** [comment]
I am having a Vercel deployment issue. @alisonjoseph Do you know how I can fix this?

**Comment by alisonjoseph** [comment]
@laurenmrice seems to be happening across all PRs, I'll take a look

**Review by alisonjoseph — APPROVED** [review]
LGTM!

File: src/pages/components/contained-list/usage.mdx [hunk 1]
```diff
@@ -193,7 +163,33 @@ other text and components on the page.
...
+
+The list header height is always fixed, while list item row heights can vary
+based on the amount of content in each row. The content in each row is flexible,
+but by default each list item row height follows the extra large, large or small
+height size for one line of content.
+
+<Row>
+<Column colLg={8}>
+
+![Row content size for on page variant](images/contained-list-usage-7.png)
```
Comment (thyhmdo) [inline src/pages/components/contained-list/usage.mdx:193]: Small detail, but it would be nice to add the fixed height indication for the list header.

Comment (laurenmrice) [reply src/pages/components/contained-list/usage.mdx:193]: I will add that for both the on-page and disclosed variant images. 👍

**Comment by laurenmrice** [comment]
— Updated images from review. @thyhmdo

**Review by thyhmdo — APPROVED** [review]
This looks great! 
One more small thing I want to point out is if it should be "On page list" of "On-page list" with a hyphen.

**Comment by laurenmrice** [comment]
— Updated "on-page" terminology to have a hyphen `on-page`. @thyhmdo

**Review by aledavila — APPROVED** [review]
Lgtm pending search

**Pull request merged**